src/realm/table.hpp

```diff
@@ -1013,7 +1033,11 @@ class LinkChain {
         }
         else {
             // Only last column in link chain is allowed to be non-link
-            throw(LogicError::type_mismatch);
+            std::string err = m_current_table->get_name();
+            err += ".";
+            err += m_current_table->get_column_name(ck);
+            err += " is not a link column";
+            throw std::runtime_error(err);
```

Comment:
Consider `throw std::runtime_error(util::format("%1.%2 is not a link column", m_current_table->get_name(), m_current_table->get_column_name(ck)));` (not blocking)

Reply:
Done